test/utils/floating_point_utils.hpp

```diff
@@ -0,0 +1,95 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace fizzy::test
+{
+/// Simple implementation of C++20's std::bit_cast.
+template <typename DstT, typename SrcT>
+DstT bit_cast(SrcT x) noexcept
+{
+    DstT z;
+    static_assert(sizeof(x) == sizeof(z));
+    __builtin_memcpy(&z, &x, sizeof(x));
+    return z;
+}
+
+/// A wrapper for floating-point types with inspection/construction/comparison utilities.
+template <typename T>
+struct FP
+{
+    static_assert(std::is_same_v<T, float> || std::is_same_v<T, double>);
```

Comment:
Hm, do you also want to add a `static_assert` for float/double being iee754 or fine as we have it in value.h and/or the limits check below?

Reply:
The `static_assert(Limits::is_iec559);` does exactly that, except only when `FP` is instantiated.